crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -1065,40 +1052,63 @@ WithItems: Vec<ast::WithItem> = {
 
 #[inline]
 WithItemsNoAs: Vec<ast::WithItem> = {
-    <location:@L> <all:OneOrMore<Test<"all">>> <end_location:@R> => {
-        all.into_iter().map(|context_expr| ast::WithItem { context_expr, optional_vars: None, range: (location..end_location).into() }).collect()
+    <all:OneOrMore<Test<"all">>> => {
+        all.into_iter().map(|context_expr| ast::WithItem {
+            range: context_expr.range(),
+            context_expr: context_expr.into(),
+            optional_vars: None,
+        }).collect()
     },
 }
 
 WithItem<Goal>: ast::WithItem = {
-    <location:@L> <context_expr: Test<Goal>> <end_location:@R> if Goal != "as" => ast::WithItem { context_expr, optional_vars: None, range: (location..end_location).into() },
-    <location:@L> <context_expr:Test<"all">> "as" <vars:Expression<"all">> <end_location:@R> => {
-        let optional_vars = Some(Box::new(set_context(vars, ast::ExprContext::Store)));
-        ast::WithItem { context_expr, optional_vars, range: (location..end_location).into() }
+    <location:@L> <context_expr:Test<Goal>> <end_location:@R> if Goal != "as" => ast::WithItem {
+        context_expr: context_expr.into(),
+        optional_vars: None,
+        range: (location..end_location).into(),
+    },
+    <location:@L> <context_expr:Test<"all">> "as" <optional_vars:Expression<"all">> <end_location:@R> => {
```

Comment:
Nit: We could probably use the range from the `context_expr` directly rather than collecting another range.

Reply:
This is leading me down some confusing path around ranges for parenthesized named expressions in with-items :sob: